lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -3707,3 +3733,75 @@ def get_test_data(delta=0.05):
     Y = Y * 10
     Z = Z * 500
     return X, Y, Z
+
+
+class _Quaternion:
+    """
+    Quaternions
+    consisting of scalar, along 1, and vector, with components along i, j, k
+    """
+
+    def __init__(self, scalar, vector):
+        self.scalar = scalar
+        self.vector = np.array(vector)
+
+    def __neg__(self):
+        return self.__class__(-self.scalar, -self.vector)
+
+    def __mul__(self, other):
+        """
+        Product of two quaternions
+        i*i = j*j = k*k = i*j*k = -1
+        """
+        return self.__class__(
+            self.scalar*other.scalar - np.dot(self.vector, other.vector),
+            self.scalar*other.vector + self.vector*other.scalar
+            + np.cross(self.vector, other.vector))
+
+    def __eq__(self, other):
+        return (self.scalar == other.scalar) and (self.vector == other.vector).all
+
+    def __repr__(self):
+        return "_Quaternion({}, {})".format(repr(self.scalar), repr(self.vector))
+
+    @classmethod
+    def from_to(cls, r1, r2):
```

Comment:
I would rename this function to something more descriptive. "from_" makes it seem like you are creating this from a "to" object.

Reply:
This "`from_`" (and "`as_`") convention is not my invention, I'm just following `numpy-quaternion` and everybody else. (And what's a "to" object, anyway?) 

As for the many valid unit quaternions that rotate a vector `a` to another vector `b` - I couldn't think of very many. I thought the options were rather limited....?

Reply:
Proposed resolution:
- Rename to `rotate_from_to()`
- Update docstring: "The quaternion for the shortest rotation from vector r1 to vector r2"
- Add extra information: how quaternion is obtained; particular case of anti-parallel vectors

Reply:
I think that sounds great to me!